packit_service/worker/handlers/koji.py

```diff
@@ -313,6 +315,24 @@ def report(self, description: str, commit_status: BaseCommitStatus, url: str):
             url=url,
         )
 
+        if (
+            not KojiOpenScanHubHelper.osh_disabled()
+            and self.db_project_event.type == ProjectEventModelType.pull_request
+            and self.build.target == "fedora-rawhide-x86_64"
+            and self.job_config.osh_diff_scan_after_copr_build
+        ):
+            try:
+                KojiOpenScanHubHelper(
+                    koji_build_helper=self.koji_build_helper,
+                    build=self.build,
+                ).handle_scan()
+            except Exception as ex:
+                # sentry_integration.send_to_sentry(ex)
+                logger.debug(
+                    f"Handling the scan raised an exception: {ex}. Skipping "
+                    f"as this is only experimental functionality for now.",
+                )
+
```

Comment:
First of all, I don't think it's a good idea to inject a scan job into reporting, for multiple reasons. It should be a separate task triggered by the same event, with a new handler in `packit_service/worker/handlers/open_scan_hub.py`. You can take a look at [this draft](https://github.com/packit/packit-service/pull/2759) for inspiration.

Reply:
I followed the same idea like [Copr](https://github.com/packit/packit-service/blob/c0ec4608a763d7fcc7a6b9b50a7457f1098ddddb/packit_service/worker/handlers/copr.py#L342).

@lachmanfrantisek Would it be possible for you to comment on it?

Reply:
> I followed the same idea like [Copr](https://github.com/packit/packit-service/blob/c0ec4608a763d7fcc7a6b9b50a7457f1098ddddb/packit_service/worker/handlers/copr.py#L342).

Good point, I didn't realize we decided to "take a shortcut" there.

Reply:
Actually, I suggested Situ to do the same we do for Copr..;)

But I agree with Nikola that it would be nice to split this functionality.